client-server-protocol/src/main/scala/zio/raft/protocol/ServerMessages.scala

```diff
@@ -0,0 +1,190 @@
+ * - Command execution results
+ * - Keep-alive responses
+ * - Server-initiated work dispatch
+ * - Error handling and leader redirection
+ */
+
+/**
+ * Base trait for all server-to-client messages.
+ */
+sealed trait ServerMessage
+
+/**
+ * Successful session creation with server-generated session ID.
+ *
+ * @param sessionId Server-generated unique session identifier
+ * @param nonce Echoed client nonce from CreateSession request
+ */
+case class SessionCreated(
+  sessionId: SessionId,
+  nonce: Nonce
+) extends ServerMessage
+
+/**
+ * Successful session resumption.
+ * Server responds to same ZeroMQ routing ID, no session ID needed.
+ *
+ * @param nonce Echoed client nonce from ContinueSession request
+ */
+case class SessionContinued(
+  nonce: Nonce
+) extends ServerMessage
+
+/**
+ * Session creation or continuation rejection.
+ * Server responds to same ZeroMQ routing ID, no session ID needed.
+ *
+ * @param reason Why the session was rejected
+ * @param nonce Echoed client nonce from request for validation
+ * @param leaderId Optional current leader ID for client redirection
+ */
+case class SessionRejected(
+  reason: RejectionReason,
+  nonce: Nonce,
+  leaderId: Option[MemberId]
+) extends ServerMessage
+
+/**
+ * Server-initiated session termination.
+ * Server responds to same ZeroMQ routing ID, no session ID needed.
+ *
+ * @param reason Why the server is closing the session
+ * @param leaderId Optional leader ID if reason is leadership-related
+ */
+case class SessionClosed(
+  reason: SessionCloseReason,
+  leaderId: Option[MemberId]
+) extends ServerMessage
+
+/**
+ * Heartbeat acknowledgment echoing client timestamp.
+ * Server responds to same ZeroMQ routing ID, no session ID needed.
+ *
+ * @param timestamp Echoed timestamp from client KeepAlive for RTT measurement
+ */
+case class KeepAliveResponse(
+  timestamp: Instant
+) extends ServerMessage
+
+/**
+ * Client request execution result.
+ * Server responds to same ZeroMQ routing ID, no session ID needed.
+ *
+ * @param requestId Echoed request ID from ClientRequest for correlation
+ * @param result Binary result data from command/query execution
+ */
+case class ClientResponse(
+  requestId: RequestId,
+  result: ByteVector
+) extends ServerMessage
+
+/**
+ * Server-initiated work dispatch to client (one-way pattern).
+ * Server targets via ZeroMQ routing ID, no session ID needed.
+ *
+ * @param requestId Unique identifier for acknowledgment correlation
+ * @param payload Binary work payload for client processing
+ * @param createdAt Timestamp when request was created
+ */
+case class ServerRequest(
+  requestId: RequestId,
+  payload: ByteVector,
+  createdAt: Instant
+) extends ServerMessage
+
+/**
+ * Client request processing error.
+ * Server responds to same ZeroMQ routing ID, no session ID needed.
+ *
+ * @param reason Error classification and details
+ * @param leaderId Optional leader ID for redirection
+ */
+case class RequestError(
```

Comment:
Let's remove this message (together with RequestErrorReason), I don't see a reason to use it at the moment.